app/controllers/movies_controller.rb

```diff
@@ -78,11 +78,16 @@ def set_movie
     end
   end #set movie
 
+  private
+
   def required_params
-    if params[:trailer].present?
-      params[:trailer] = params[:trailer].gsub('https://www.youtube.com/watch?v=', '')
-    end
+    params[:trailer] = parse_youtube_url(params[:trailer]) if params[:trailer].present?
     params.permit(:trailer)
   end
 
+  def parse_youtube_url(trailer)
+    return trailer unless trailer.include?('youtube.com')
+    uri = Addressable::URI.parse(trailer)
+    uri.query_values["v"] if uri.query_values
+  end
```

Comment:
When I was reading this, I was a little confused about what was being returned. In one case, we're returning a whole url, in another, we're returning just a youtube_id, but it took me a beat to get there. When I rearrange the two methods like this, it's a little bit more clear _to me_. Of course, it's your call.

```ruby
  def required_params
    trailer_url = params[:trailer]
    params[:trailer] = trailer_url.include?('youtube.com') ? youtube_id : trailer_url
    params.permit(:trailer)
  end

  def youtube_id
    uri = Addressable::URI.parse(params[:trailer])
    uri.query_values['v'] if uri.query_values
  end

```

Reply:
👍 Ooh, me likey. Much better!